lib/ansible/plugins/inventory/ini.py

```diff
@@ -106,8 +123,21 @@ def __init__(self) -> None:
         self._origin: Origin | None = None
 
     def verify_file(self, path):
-        # hardcode exclusion for TOML to prevent partial parsing of things we know we don't want
-        return super().verify_file(path) and os.path.splitext(path)[1] != '.toml'
+
+        valid = super().verify_file(path)
+
+        p = Path(path)
+        allowed = self.get_option('allowed_extensions')
+        if valid:
+            if allowed:
+                if p.suffix[1:] not in allowed:
+                    valid = False
+            elif p.suffix[1:] != 'ini':
+                self.display.deprecated(
```

Comment:
The warning should be deferred until a successful exit from `parse`, otherwise it will show up for every file `ini` sees but does not parse (e.g., the original `.toml` case).

Reply:
roger